### PR TITLE
Send acceptance email by ID

### PR DESCRIPTION
**Pull request by Jeremyyang920**

### CODE REVIEW

File: app/server/controllers/UserController.js [hunk 1]
```diff
@@ -707,12 +707,32 @@ UserController.checkOutById = function(id, user, callback){
 };
 
 
+
```
Comment (madhavgharmalkar) [inline app/server/controllers/UserController.js]: why space yo

File: app/server/services/email.js [hunk 2]
```diff
@@ -194,4 +194,41 @@ controller.sendPasswordChangedEmail = function(email, callback){
 
 };
 
+/**
+ * Send the acceptance email to the participant.
+ * @param  {[type]}   email    [description]
+ * @param  {Function} callback [description]
+ */
+controller.sendAcceptanceEmail = function(email, callback){
```
Comment (chriswongtv) [inline app/server/services/email.js:202]: same as above

File: app/server/services/email.js [hunk 3]
```diff
@@ -194,4 +194,41 @@ controller.sendPasswordChangedEmail = function(email, callback){
...
+    body: 'Please sign into your dashboard to confirm your spot.',
+  };
+
+  /**
+   * Eamil-verify takes a few template values:
+   * {
+   *   verifyUrl: the url that the user must visit to verify their account
+   * }
+   */
+  sendOne('email-basic', options, locals, function(err, info){
```
Comment (chriswongtv) [inline app/server/services/email.js:220]: same as above

File: scripts/acceptUsers.js [hunk 4]
```diff
@@ -1,17 +1,19 @@
 require('dotenv').load();
 var mongoose        = require('mongoose');
-var database        = process.env.DATABASE || "mongodb://localhost:27017";
+var database        = process.env.DATABASE || "mongodb://develop:HackUCI2018@cluster0-shard-00-00-j5esm.mongodb.net:27017,cluster0-shard-00-01-j5esm.mongodb.net:27017,cluster0-shard-00-02-j5esm.mongodb.net:27017/test?ssl=true&replicaSet=Cluster0-shard-0&authSource=admin";
```
Comment (madhavgharmalkar) [inline scripts/acceptUsers.js]: don't hardcode the uri sting

File: app/server/services/email.js [hunk 5]
```diff
@@ -194,4 +194,41 @@ controller.sendPasswordChangedEmail = function(email, callback){
...
+    if (err){
```
Comment (chriswongtv) [inline app/server/services/email.js:221]: same as above

File: scripts/acceptUsers.js [hunk 6]
```diff
@@ -1,17 +1,19 @@
...
 var user = { email: process.env.ADMIN_EMAIL };
-
+/**
+ * Send the acceptance email and updates participant profile.
+ */
 var userArray = require('fs').readFileSync('accepted.txt').toString().split('\n');
 var count = 0;
 userArray.forEach(function (id) {
-  UserController.admitUser( id, user, function() {
+  UserController.sendAcceptanceEmailById( id, function() {
```
Comment (madhavgharmalkar) [inline scripts/acceptUsers.js]: no space after (,

also for legacy purposes lets leave the "admit user" and add this function after

**Review by madhavgharmalkar — CHANGES_REQUESTED** [review]

File: app/server/services/email.js [hunk 7]
```diff
@@ -194,4 +194,41 @@ controller.sendPasswordChangedEmail = function(email, callback){
...
+      console.log(err);
+    }
+    if (info){
```
Comment (chriswongtv) [inline app/server/services/email.js:224]: nitpick: leave a space between `if` statements + the above

File: app/server/services/email.js [hunk 8]
```diff
@@ -194,4 +194,41 @@ controller.sendPasswordChangedEmail = function(email, callback){
...
+      console.log(info.message);
+    }
+    if (callback){
```
Comment (chriswongtv) [inline app/server/services/email.js:227]: same as above

File: scripts/acceptUsers.js (hunk 4 above)
Comment (chriswongtv) [inline scripts/acceptUsers.js]: umm... is this meant to be public?

File: app/server/controllers/UserController.js [hunk 9]
```diff
@@ -707,12 +707,32 @@ UserController.checkOutById = function(id, user, callback){
...
 /**
  * [ADMIN ONLY]
  */
+ /**
+  * Send the acceptance email to the participant by their ID.
+  * @param  {[type]}   ID    [description]
+  * @param  {Function} callback [description]
+  */
+UserController.sendAcceptanceEmailById = function(id, callback){
```
Comment (chriswongtv) [inline app/server/controllers/UserController.js:718]: nitpick: leave an empty space between the parentheses and the braces

File: app/server/controllers/UserController.js [hunk 10]
```diff
@@ -707,12 +707,32 @@ UserController.checkOutById = function(id, user, callback){
...
+   User.findOne(
+     {
+       _id: id,
+       verified: true
+     },
+     function(err, user){
```
Comment (chriswongtv) [inline app/server/controllers/UserController.js:724]: same as above

File: app/server/controllers/UserController.js [hunk 11]
```diff
@@ -707,12 +707,32 @@ UserController.checkOutById = function(id, user, callback){
...
+       if (err || !user){
```
Comment (chriswongtv) [inline app/server/controllers/UserController.js:725]: same as above

**Review by madhavgharmalkar — APPROVED** [review]
yay

**Pull request merged**